### PR TITLE
Add org.siril.Siril

### PR DESCRIPTION
**Pull request by lock042**
### Please confirm your submission meets all the criteria

- [X] Please describe the application briefly.
  Siril is an image processing tool specially tailored for noise reduction and improving the signal/noise ratio of an image from multiple captures, as required in astronomy.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X] I have read the and followed all the [Submission and licence requirements][reqs].
- [X] I have [built][build] and tested the submission locally.
- [X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**

This submission is a resubmission to change AP-ID in order to verify the package against https://siril.org

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

**Comment by lock042** [comment]
@hfiguiere : I ping you to let you know. I am resubmitting the package to allow verification against our website: https://siril.org

File: org.siril.Siril.json [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+    "rename-mime-file": "siril.xml",
+    "rename-mime-icons": [
+        "text-x-seq"
+    ],
+    "provides": {
+        "id": "org.free_astro.siril"
+    },
+    "replaces": {
+        "id": "org.free_astro.siril"
+    },
```
Comment (bbhtt) [inline org.siril.Siril.json:17]: This is supposed to be in your metainfo file not here 

https://gitlab.com/free-astro/siril/-/blob/master/platform-specific/linux/org.free_astro.siril.appdata.xml?ref_type=heads

File: org.siril.Siril.json [hunk 2]
```diff
@@ -0,0 +1,288 @@
...
+            "cleanup": [
+                "/include",
+                "/lib/pkgconfig"
+            ],
+            "sources": [
+                {
+                    "type": "git",
+                    "url": "https://github.com/FFMS/ffms2.git",
+		    "commit": "a8ed11c21eea1f2ba4c84d1ea0d2cadcc8ae9422"
+                }
```
Comment (bbhtt) [inline org.siril.Siril.json:242]: indent is lost

File: org.siril.Siril.json [hunk 3]
```diff
@@ -0,0 +1,288 @@
+{
+    "app-id": "org.siril.Siril",
```
Comment (bbhtt) [inline org.siril.Siril.json:2]: Can you link the Gitlab somewhere on https://siril.org/ ?

File: org.siril.Siril.json [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+    },
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=host",
+        /* Needed for gvfs to work */
+        "--talk-name=org.gtk.vfs.*",
+        "--filesystem=xdg-run/gvfs",
```
Comment (bbhtt) [inline org.siril.Siril.json:26]: Is this needed ? According to https://docs.flatpak.org/en/latest/sandbox-permissions.html#gvfs-access

> Typical GNOME and GTK applications should use:
> --talk-name=org.gtk.vfs.*
> --filesystem=xdg-run/gvfsd

File: org.siril.Siril.json [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+    },
+    "replaces": {
+        "id": "org.free_astro.siril"
+    },
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
+        "--filesystem=host",
```
Comment (bbhtt) [inline org.siril.Siril.json:23]: why is host needed?

File: org.siril.Siril.json [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+        "id": "org.free_astro.siril"
+    },
+    "replaces": {
+        "id": "org.free_astro.siril"
+    },
+    "finish-args": [
+        "--share=ipc",
+        "--share=network",
+        "--socket=x11",
+        "--socket=wayland",
```
Comment (bbhtt) [inline org.siril.Siril.json:22]: It can't be x11 or wayland both. It's only x11 or fallback-x11+wayland.

Comment (bbhtt) [reply org.siril.Siril.json:22]: See the note in https://docs.flatpak.org/en/latest/sandbox-permissions.html#standard-permissions

File: org.siril.Siril.json [hunk 7]
```diff
@@ -0,0 +1,288 @@
...
+            "config-opts": [
+                "--enable-reentrant"
+            ],
+            "make-args": ["shared"],
+            "cleanup": [
+                "/include",
+                "/lib/*.a",
+                "/lib/*.la",
+                "/lib/pkgconfig"
+            ],
```
Comment (bbhtt) [inline org.siril.Siril.json:42]: Can you combine the common cleanups into a toplevel one ?

`/lib/*.a` -> `"*.a" and `/lib/*.la` -> `"*.la"`

File: org.siril.Siril.json (hunk 4 above)
Comment (hfiguiere) [reply org.siril.Siril.json:26]: I remember checking back then these were the right one.

File: org.siril.Siril.json [hunk 8]
```diff
@@ -0,0 +1,288 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://mirror.sergal.org/gnu/gsl/gsl-2.7.1.tar.gz",
+                    "sha256": "dcb0fbd43048832b757ff9942691a8dd70026d5da0ff85601e52687f6deeb34b"
+                }
+            ],
+            "cleanup": [
+                "/bin",
+                "/include",
```
Comment (bbhtt) [inline org.siril.Siril.json:89]: Put everything except /bin in toplevel cleanup.

File: org.siril.Siril.json [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+            "sources": [
+                {
+                    "type": "archive",
+                    "url": "http://fftw.org/fftw-3.3.10.tar.gz",
+                    "sha256": "56c932549852cddcfafdab3820b0200c7742675be92179e59e6215b340e26467"
+                }
+            ],
+            "cleanup": [
+                "/bin",
+                "/include",
```
Comment (bbhtt) [inline org.siril.Siril.json:132]: Same here. Put everything except /bin in toplevel cleanup.

File: org.siril.Siril.json (hunk 1 above)
Comment (lock042) [reply org.siril.Siril.json]: Fixed in 57d8a6a

File: org.siril.Siril.json (hunk 2 above)
Comment (lock042) [reply org.siril.Siril.json]: fixed in 57d8a6a

File: org.siril.Siril.json (hunk 9 above)
Comment (lock042) [reply org.siril.Siril.json:125]: 57d8a6a

File: org.siril.Siril.json (hunk 7 above)
Comment (lock042) [reply org.siril.Siril.json]: fixed in 57d8a6a

File: org.siril.Siril.json (hunk 5 above)
Comment (lock042) [reply org.siril.Siril.json:16]: If I well remember, without it there are some permission issues.

File: org.siril.Siril.json (hunk 3 above)
Comment (lock042) [reply org.siril.Siril.json:2]: The gitlab is linked in the FAQ: https://siril.org/faq/#how-can-i-contact-sirils-team

**Comment by lock042** [comment]
Need I update the name of all files in the app?
*.desktop
*.xml
and icons ?

File: org.siril.Siril.json (hunk 3 above)
Comment (bbhtt) [reply org.siril.Siril.json:2]: thanks

**Comment by bbhtt** [comment]
> Need I update the name of all files in the app?

Yes everything must match the new ID. also any metainfo tags where the ID is used like launchable etc.

**Comment by lock042** [comment]
> > Need I update the name of all files in the app?
> 
> Yes everything must match the new ID. also any metainfo tags where the ID is used like launchable etc.

OK. This is very annoying just to get the Verify status.

**Comment by lock042** [comment]
@bbhtt @hfiguiere : is org.siril.Siril ok then, to match with https://siril.org ?

**Comment by bbhtt** [comment]
> OK. This is very annoying just to get the Verify status.

If the domain changes the app id needs to change too. There isn't a way around that. 

> @bbhtt @hfiguiere : is org.siril.Siril ok then, to match with https://siril.org/ ?

Yes

**Comment by lock042** [comment]
@bbhtt :  I've fixed all the issues I believe.

**Comment by bbhtt** [comment]
bot, build org.siril.Siril

**Comment by bbhtt** [comment]
shared-modules gitmodule is missing

**Comment by lock042** [comment]
> shared-modules gitmodule is missing

So sorry, I forgot to push it. Fixed

**Comment by bbhtt** [comment]
You can start the build.

bot, build org.siril.Siril

**Comment by lock042** [comment]
Hum. I'm not sure to understand these errors:

```
{
    "errors": [
        "appstream-failed-validation"
    ],
    "appstream": [
        "E: org.siril.Siril:89: tag-duplicated provides"
    ],
    "info": [
        "appstream-failed-validation: Metainfo file /tmp/tmpqdbg5y96/metainfo/org.siril.Siril.metainfo.xml has failed validation. Please see the errors in appstream block"
    ],
    "message": "Please consult the documentation at https://docs.flathub.org/docs/for-app-authors/linter"
}
```

**Comment by lock042** [comment]
bot, build org.siril.Siril

**Comment by lock042** [comment]
@bbhtt : That looks good now. Sorry for the noise :)

**Comment by lock042** [comment]
Preparing EOL merge here: https://github.com/flathub/org.free_astro.siril/pull/55

**Pull request closed without merging**